compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4008,9 +4008,18 @@ public void visit(BLangBinaryExpr binaryExpr) {
 
         checkDecimalCompatibilityForBinaryArithmeticOverLiteralValues(binaryExpr);
 
+        BType nilLiftType = null;
+
+        if (isNullableBinaryExpr(binaryExpr)) {
```

Comment:
Let's extract `isNullableBinaryExpr(binaryExpr)` out to a variable since this is used in multiple places.